### PR TITLE
fix ast_to_html with empty elements

### PR DESCRIPTION
**Pull request by Zurga**
While creating documentation that contained `<i></i>` inside a `<span>` element, the formatter was not able to cope with this html. See the included test.

### CODE REVIEW

File: lib/ex_doc/formatter/html.ex [hunk 1]
```diff
@@ -128,7 +128,7 @@ defmodule ExDoc.Formatter.HTML do
   def ast_to_html(binary) when is_binary(binary), do: Templates.h(binary)
 
   def ast_to_html({tag, attrs, ast, %{verbatim: true}}) do
-    [text] = ast
+    text = (ast != [] && Enum.join(ast, "")) || ""
```
Comment (josevalim) [inline lib/ex_doc/formatter/html.ex:131]:
```suggestion
    text = Enum.join(ast, "")
```

This will work for one element and no element ASTs. :)

Comment (Zurga) [reply lib/ex_doc/formatter/html.ex:131]: Yep!

**Review by josevalim — APPROVED** [review]

File: lib/ex_doc/formatter/html.ex [hunk 2]
```diff
@@ -128,7 +128,7 @@ defmodule ExDoc.Formatter.HTML do
   def ast_to_html(binary) when is_binary(binary), do: Templates.h(binary)
 
   def ast_to_html({tag, attrs, ast, %{verbatim: true}}) do
-    [text] = ast
+    text = Enum.join(ast, "")
     ["<#{tag}", ast_attributes_to_html(attrs), ">", text, "</#{tag}>"]
```
Comment (wojtekmach) [inline lib/ex_doc/formatter/html.ex:132]: this function returns an iolist so there's no need to join the elements:

```suggestion
    ["<#{tag}", ast_attributes_to_html(attrs), ">", ast, "</#{tag}>"]
```

File: test/ex_doc/formatter/html_test.exs [hunk 3]
```diff
@@ -527,5 +527,17 @@ defmodule ExDoc.Formatter.HTMLTest do
...
+    test "handles empty elements" do
+      assert """
+             <span>
+             <i></i>
+             </span>
+             """
+             |> ExDoc.Markdown.Earmark.to_ast([])
+             |> HTML.ast_to_html()
+             |> IO.iodata_to_binary() ==
+               "<span><i></i></span>"
```
Comment (wojtekmach) [inline test/ex_doc/formatter/html_test.exs:540]: This test passes on master without your change, could you rebase and try again?

I was able to see a failure with such markdown though: `<i></i>`, If you can verify it, could you test that instead ?

Comment (Zurga) [reply test/ex_doc/formatter/html_test.exs:540]: Ah ok, my test case initially contained two `<i></i>` and it failed. After fixing that I decided to remove one of the elements and it still worked but I did not test that on `master`. I will make the test more explicit.

File: lib/ex_doc/formatter/html.ex (hunk 2 above)
Comment (Zurga) [reply lib/ex_doc/formatter/html.ex:132]: You are right

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**